File: Beispiele/Java2Kotlin/src/main/kotlin/javakotlinworkshop/java/EqualityDemo.java
package javakotlinworkshop.java;

import java.nio.charset.Charset;

public class EqualityDemo {

    public static void main(String[] args) {
        var a = "Hallo KKON";
        var b = new String(a.getBytes(Charset.defaultCharset()),
                Charset.defaultCharset());

        // referential equality
        System.out.println(String.format("a == b: %b", a == b));

        // structural equality
        System.out.println(String.format("a.equals(b): %b", a.equals(b)));
    }
}

File: Beispiele/Java2Kotlin/src/main/kotlin/javakotlinworkshop/kotlin/EqualityDemo.kt
package javakotlinworkshop.kotlin

import java.nio.charset.Charset

fun main(args: Array<String>) {

    val a = "Hallo KKON"
    val b = String(a.toByteArray(Charset.defaultCharset()),
            Charset.defaultCharset())

    // referential equality
    println(String.format("a === b: %b", a === b))

    // structural equality
    println(String.format("a == b: %b", a == b))
}